types/mempool.go

```diff
@@ -1,14 +1,19 @@
 package types
 
 import (
+	"crypto/sha256"
 	"errors"
 	"fmt"
 )
 
-var (
-	// ErrTxInCache is returned to the client if we saw tx earlier
-	ErrTxInCache = errors.New("tx already exists in cache")
-)
+// ErrTxInCache is returned to the client if we saw tx earlier
+var ErrTxInCache = errors.New("tx already exists in cache")
+
+// TxKeySize defines the size of the transaction's key used for indexing.
+const TxKeySize = sha256.Size
```

Comment:
We probably don't need to export this anymore—it's not used anywhere but here. I'd just inline it in the type.

```go
type TxKey [sha256.Size]byte
```

or if not then I'd unexport the const.